app/code/Magento/CatalogImportExport/Model/Import/Product/LinkProcessor.php

```diff
@@ -0,0 +1,340 @@
+                    continue;
+                }
+
+                $sku = $rowData[Product::COL_SKU];
+
+                $productId = $this->skuProcessor->getNewSku($sku)[$this->_productEntityLinkField];
+                $productLinkKeys = $this->fetchExistingLinks($productId);
+
+                foreach ($this->_linkNameToId as $linkName => $linkId) {
+                    $productIds[] = $productId;
+
+                    $linkSkus = $this->getLinkSkus($rowData, $linkName);
+                    if ($linkSkus === false) {
+                        continue;
+                    }
+
+                    $linkPositions = $this->getLinkPositions($rowData, $linkName);
+
+                    foreach ($linkSkus as $linkedKey => $linkedSku) {
+                        $linkedSku = trim($linkedSku); //NOTE: why is trimming happening here and not for all cols in a general place?
+                        if ( ! $this->isProperLink($linkedSku, $sku)) {
+                            continue;
+                        }
+
+                        $linkedId = $this->getLinkedId($linkedSku);
+
+                        if ($this->checkForEmptyLinkId($linkedId, $sku, $productId, $linkedSku)) {
+                            continue;
+                        }
+
+                        $linkKey = "{$productId}-{$linkedId}-{$linkId}";
+                        if (empty($productLinkKeys[$linkKey])) {
+                            $productLinkKeys[$linkKey] = $nextLinkId;
+                        }
+                        if ( ! isset($linkRows[$linkKey])) {
+                            $linkRows[$linkKey] = [
+                                'link_id' => $productLinkKeys[$linkKey],
+                                'product_id' => $productId,
+                                'linked_product_id' => $linkedId,
+                                'link_type_id' => $linkId,
+                            ];
+                        }
+                        if ( ! empty($linkPositions[$linkedKey])) {
+                            $positionRows[] = [
+                                'link_id' => $productLinkKeys[$linkKey],
+                                'product_link_attribute_id' => $positionAttrId[$linkId],
+                                'value' => $linkPositions[$linkedKey],
+                            ];
+                        }
+                        $nextLinkId++;
+                    }
+                }
+            }
+
+            $this->deleteExistingLinks($productIds);
+            $this->insertNewLinks($linkRows, $positionRows);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param $linkId
+     * @param $sku
+     * @param $productId
+     * @param string $linkedSku
+     */
+    protected function checkForEmptyLinkId($linkedId, $sku, $productId, string $linkedSku): bool
+    {
+        if ($linkedId == null) {
+            // Import file links to a SKU which is skipped for some reason,
```

Comment:
I would be tempted to add this note in the method comment rather than in the code comment. What do you think?

Reply:
okay